qgym/envs/initial_mapping/initial_mapping.py

```diff
@@ -195,22 +199,22 @@ def __init__(  # pylint: disable=too-many-arguments
             networkx.generators.lattice.grid_graph.html#grid-graph
         """
         # Check user input and parse it to a uniform format
-        interaction_graph_edge_probability = check_real(
-            interaction_graph_edge_probability,
-            "interaction_graph_edge_probability",
-            l_bound=0,
-            u_bound=1,
-        )
         connection_graph = parse_connection_graph(
             connection_graph, connection_graph_matrix, connection_grid_size
         )
 
+        if graph_generator is None:
+            graph_generator = BasicGraphGenerator(len(connection_graph), seed=self.rng)
+        else:
+            check_instance(graph_generator, "graph_generator", GraphGenerator)
+            if graph_generator.finite:
+                raise ValueError("'graph_generator' should not be an infinite iterator")
```

Comment:
Since Reinforcement Learning is relatively data inefficient, an infinite training set should be advantageous. This can also be done by randomly drawing from a finite set of training samples (or looping over them).

Reply:
Okay, makes sense!